dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/reader/EntrezXMLReader.java

```diff
@@ -79,82 +80,123 @@ private static SAXParserFactory createParserFactory()
      */
     private final Map<String, String> summaryMap;
 
-    private final Stack<String> elementStack = new Stack<String>();
-    private boolean isEnsembl = false;
-    private String currentId;
     private StringBuilder currentValue;
 
+    private GeneRefDb currentGeneRef;
+    private EntrezGene currentEntrezGene;
+    private boolean isGeneRefSectionActive = false;
+
     /**
      * Dependencies
      */
     @NonNull
     public List<String> ids = new ArrayList<String>();
 
-    @Override
-    public void startElement(String uri, String localName, String qName, Attributes attributes) {
-      if ("Entrezgene_summary".equals(qName)) {
-        currentValue = new StringBuilder();
-      }
-
-      this.elementStack.push(qName);
+    private void resetValueBuffer() {
+      currentValue = new StringBuilder();
     }
 
     @Override
-    public void endElement(String uri, String localName, String qName) {
-      if ("Entrezgene_summary".equals(qName)) {
-        summaryMap.put(currentId, currentValue.toString());
+    public void startElement(String uri, String localName, String qName, Attributes attributes) {
+      switch (qName) {
```

Comment:
since alot of cases are just calling `resetValueBuffer`, could potentially group all the cases that do this together and just have one `resetValueBuffer` and one `break` statement. but fine like this too

Reply:
If I switch to using an `if else` construct I should be able to do that